packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx

```diff
@@ -206,18 +206,27 @@ const DetailPageImpl = memo(function DetailPageImpl() {
     [jumpToPageBlock, docCollection.id, openPage, jumpToTag, workspace.id]
   );
 
+  const [container, setContainer] = useState<HTMLElement | null>(null);
```

Comment:
why not useRef

Reply:
in the domain of useHasScrollTop there is no knowledge when will the ref will be set. I think alternatively we can use the ref-callback solution, which should be better without additional rerendering.